CONTRIBUTING.md

```diff
@@ -76,6 +76,10 @@ Maintainers are:
 ### PR Checklist
 
 Before your PR can be merged it must meet the following criteria:
+* PR Desciption should be informative, explaining what is in it and a high level design in case of new feature
+* Commits should be split logically with a single clear purpose for each.
+* Commits title should be concise and clear while commit message should be as informative as possible.
+* New changes done (after review for example) should be squashed to relevant commit.
```

Comment:
It depends cause if a second reviewer is going over the PR after more changes were introduced then he will comment on the commit and then in the newer "updates" commit he might see that everything that he invested time on reviewing was already fix or that his review is already unrelevant

Reply:
I can see why this is not trivial... harder to review vs harder to make sense of later
but until we start not squashing I lean towards dropping this

Reply:
> If the majority thinks we should drop it then I will,

@ShellyKa13 the burden is on you to convince the majority.  Not the other way around.  This PR will rot until a significant number of active contributors come forward in support.

Reply:
I support @ShellyKa13 on this point.  This was a principal part of her arguments.  The commit log of CDI should be clean and changing the same exact lines in multiple commits introduces noise and makes going back to look at individual commits useless.  The thrust of this PR is to "Make `git-log` readable again".

Reply:
I think you can compare changes between force-pushes using the github UI so even if a PR author squashes their review comment driven changes into the set of existing commits you can still see only what's been changed since the last round of review.